src/main/scala/com/github/pathikrit/dijon/package.scala

```diff
@@ -135,71 +131,93 @@ package object dijon {
     override def hashCode: Int = underlying.hashCode
   }
 
-  def compact(json: SomeJson): String = writeToString[SomeJson](json)
+  def compact(json: SomeJson): String = new String(writeToArray[SomeJson](json), UTF_8)
 
-  def pretty(json: SomeJson): String = writeToString[SomeJson](json, WriterConfig.withIndentionStep(2))
+  def pretty(json: SomeJson): String = new String(writeToArray[SomeJson](json, prettyConfig), UTF_8)
 
-  def parse(s: String): SomeJson = readFromString[SomeJson](s)
+  def parse(s: String): SomeJson = readFromArray[SomeJson](s.getBytes(UTF_8))
 
   implicit class JsonStringContext(val sc: StringContext) extends AnyVal {
     def json(args: Any*): SomeJson = parse(sc.s(args: _*))
   }
 
   implicit val codec: JsonValueCodec[SomeJson] = new JsonValueCodec[SomeJson] {
-    override def decodeValue(in: JsonReader, default: SomeJson): SomeJson = (in.nextToken(): @switch) match {
```

Comment:
Interesting that if-else is better here, I'm guessing this was compiling to lookupswitch instead of tableswitch?